src/app/api/handle-chapter-request/route.ts

```diff
@@ -67,6 +70,52 @@ export const POST = async (request: NextRequest) => {
             approved: "APPROVED",
           },
         });
+
+        const baseFolder = process.env.GOOGLE_BASEFOLDER; // TODO: make env variable
+        const fileMetadata = {
+          name: [`${chapterRequest.university}-${createdChapter.id}`],
+          mimeType: "application/vnd.google-apps.folder",
+          parents: [baseFolder],
+        };
+        const fileCreateData = {
+          resource: fileMetadata,
+          fields: "id",
+        };
+
+        const { access_token, refresh_token } = (await prisma.account.findFirst(
+          {
+            where: {
+              userId: session.user.id,
+            },
+          }
+        )) ?? { access_token: null };
+        const auth = new google.auth.OAuth2({
+          clientId: env.GOOGLE_CLIENT_ID,
+          clientSecret: env.GOOGLE_CLIENT_SECRET,
+        });
+        auth.setCredentials({
+          access_token,
+          refresh_token,
+        });
+        const service = google.drive({
+          version: "v3",
+          auth,
+        });
+
+        const file = await (
+          service as NonNullable<typeof service>
+        ).files.create(fileCreateData);
+        const googleFolderId = (file as any).data.id;
+
+        await prisma.chapter.update({
```

Comment:
I wonder if it's necessary to include the new chapter ID as part of the folder name. We currently don't have any constraints that uni name is unique. What do you guys think?